### PR TITLE
[#642] HIRS_AttestationCA spotbug fixes

### PR DESCRIPTION
**Pull request by cyrus-dev**
Initial set of changes for #642 that are fixes for HIRS_AttestationCA. In addition, this and the following pushes will have changes for #651 because spot bugs doesn't like how Lombok handles hashCode and equals

### CODE REVIEW

File: HIRS_AttestationCA/src/main/java/hirs/attestationca/persist/entity/userdefined/certificate/attributes/ComponentIdentifier.java [hunk 1]
```diff
@@ -121,7 +120,7 @@ public ComponentIdentifier(final DERUTF8String componentManufacturer,
         this.componentRevision = componentRevision;
         this.componentManufacturerId = componentManufacturerId;
         this.fieldReplaceable = fieldReplaceable;
-        this.componentAddress = componentAddress;
+        this.componentAddress = componentAddress.stream().toList();
```
Comment (chubtub) [inline HIRS_AttestationCA/src/main/java/hirs/attestationca/persist/entity/userdefined/certificate/attributes/ComponentIdentifier.java:123]: Is this to copy the values of componentAddress into this.componentAddress?

**Review by chubtub — COMMENTED** [review]
Changes look good, see the questions I had for a few of the classes.

**Review by chubtub — APPROVED** [review]
Changes look good

File: HIRS_AttestationCA/src/main/java/hirs/attestationca/persist/entity/userdefined/certificate/attributes/V2/ComponentIdentifierV2.java [hunk 2]
```diff
@@ -200,6 +199,24 @@ public boolean isVersion2() {
         return true;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        if (!super.equals(o)) return false;
+        ComponentIdentifierV2 that = (ComponentIdentifierV2) o;
+        return Objects.equals(componentClass, that.componentClass)
```
[the remark this answers is not in the document]
Comment (cyrus-dev) [reply HIRS_AttestationCA/src/main/java/hirs/attestationca/persist/entity/userdefined/certificate/attributes/V2/ComponentIdentifierV2.java:208]: auto generated code

File: HIRS_AttestationCA/src/main/java/hirs/attestationca/persist/entity/userdefined/rim/BaseReferenceManifest.java [hunk 3]
```diff
@@ -331,6 +336,43 @@ private Document removeXMLWhitespace(final ByteArrayInputStream byteArrayInputSt
         return document;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        if (!super.equals(o)) return false;
+        BaseReferenceManifest that = (BaseReferenceManifest) o;
+        return swidCorpus == that.swidCorpus && Objects.equals(swidName, that.swidName)
```
[the remark this answers is not in the document]
Comment (cyrus-dev) [reply HIRS_AttestationCA/src/main/java/hirs/attestationca/persist/entity/userdefined/rim/BaseReferenceManifest.java:345]: auto generated code

File: HIRS_AttestationCA/src/main/java/hirs/attestationca/persist/entity/userdefined/rim/SupportReferenceManifest.java [hunk 4]
```diff
@@ -119,4 +120,18 @@ public Collection<TpmPcrEvent> getEventLog() {
...
         return !this.isSwidSupplemental() && !this.isSwidPatch();
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        if (!super.equals(o)) return false;
+        SupportReferenceManifest that = (SupportReferenceManifest) o;
+        return pcrHash == that.pcrHash && updated == that.updated && processed == that.processed;
```
[the remark this answers is not in the document]
Comment (cyrus-dev) [reply HIRS_AttestationCA/src/main/java/hirs/attestationca/persist/entity/userdefined/rim/SupportReferenceManifest.java:130]: auto generated code

**Pull request merged**